### PR TITLE
implemented missing 2 features

### PR DESCRIPTION
**Pull request by Marchuck**
Hi there!
I faced two features missing in this library, so I decided to implement this on my own:

I have added additional listener(similar to OnDateSelectedListener) but has additional callback for clicked DayView(can be used for displaying popups or other tooltips).

Also - I really needed feature regarding coloring particular days(for example holidays) using DayViewDecorator - I have implemented additional setter.

Hope everything is readable

Best

Łukasz Marczak

### CODE REVIEW

File: library/src/main/java/com/prolificinteractive/materialcalendarview/DayViewFacade.java [hunk 1]
```diff
@@ -36,6 +37,13 @@ public void setBackgroundDrawable(@NonNull Drawable drawable) {
         isDecorated = true;
     }
 
+    public void setTextColor(Integer textColor) {
+        if (textColor!=null){
```
Comment (tevjef) [inline library/src/main/java/com/prolificinteractive/materialcalendarview/DayViewFacade.java]: Minor code style issue:

```java
 if (textColor != null) {
```

File: build.gradle [hunk 2]
```diff
@@ -1,15 +1,17 @@
 buildscript {
     repositories {
         jcenter()
+        google()
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:2.4.0-alpha7'
+        classpath 'com.android.tools.build:gradle:3.1.1'
```
Comment (tevjef) [inline build.gradle]: Please rebase on the latest branch to get the most recent versions of dependencies.

File: library/src/main/java/com/prolificinteractive/materialcalendarview/MaterialCalendarView.java [hunk 3]
```diff
@@ -442,7 +444,8 @@ private void updateUi() {
      * @see #SELECTION_MODE_RANGE
      */
     public void setSelectionMode(final @SelectionMode int mode) {
-        final @SelectionMode int oldMode = this.selectionMode;
+        final @SelectionMode
+        int oldMode = this.selectionMode;
```
Comment (tevjef) [inline library/src/main/java/com/prolificinteractive/materialcalendarview/MaterialCalendarView.java]: Minor code style issue. This should be one line.
``` java
final @SelectionMode int oldMode = this.selectionMode
```

**Comment by Marchuck** [comment]
@tevjef I resolved issues you noticed. Let me know if I should improve something else.

File: build.gradle [hunk 4]
```diff
@@ -17,6 +18,7 @@ allprojects {
     repositories {
         google()
         jcenter()
+        google()
```
Comment (barnhill) [inline build.gradle]: googles maven repo is already in the hierarchy

**Comment by quentin41500** [comment]
@tevjef Could we just combine `OnDatePickedListener` and `OnDateSelectedListener` ?

**Comment by tevjef** [comment]
@quentin41500 Can track this for 2.0?

**Pull request closed without merging**